Tests/NIOTests/ByteBufferTest.swift

```diff
@@ -233,6 +233,40 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(5, buf.writerIndex)
     }
 
+    func testWithMutableWritePointerWithMinimumSpecifiedAdjustsCapacity() {
+        XCTAssertEqual(0, buf.writerIndex)
+        XCTAssertEqual(512, buf.capacity)
+
+        var bytesWritten = buf.writeWithUnsafeMutableBytes{ (_: UnsafeMutableRawBufferPointer) in return 256 }
+        XCTAssertEqual(256, bytesWritten)
+        XCTAssertEqual(256, buf.writerIndex)
+        XCTAssertEqual(512, buf.capacity)
+
+        bytesWritten += buf.writeWithUnsafeMutableBytes(minimumWritableBytes: 1024) { (_: UnsafeMutableRawBufferPointer) in return 1024 }
```

Comment:
Can these callbacks validate the pointer has the correct minimum size?

Reply:
Good idea! Added.